pkg/controller/storagecluster/storagecluster.go

```diff
@@ -643,6 +643,28 @@ func (c *Controller) deleteStorageCluster(
 		msg := fmt.Sprintf("Failed to cleanup Stork. %v", err)
 		k8s.WarningEvent(c.recorder, cluster, util.FailedComponentReason, msg)
 	}
+
+	if err := c.removeMigrationLabels(); err != nil {
```

Comment:
I think we should do this before the finalizer is removed. That way if the node update fails, we retry again.